src/dataflow/src/source/timestamp.rs

```diff
@@ -746,26 +743,22 @@ impl Codec for SourceTimestamp {
         "protobuf[SourceTimestamp]".into()
     }
 
-    fn encode<E: for<'a> Extend<&'a u8>>(&self, buf: &mut E) {
+    fn encode<B: BufMut>(&self, buf: &mut B) {
         ProtoSourceTimestamp::from(self)
-            .write_to_writer(&mut ExtendWriteAdapter(buf))
-            .expect("infallible for ExtendWriteAdapter")
+            .encode(buf)
+            .expect("infallible")
```

Comment:
the prost docs declare that encoding only returns an error if the provided buffer had insufficient capacity. I really hope you don't have to eat a CI cycle over this, but I'd much prefer if all these expects were something to the effect of `.expect("provided buffer had sufficient capacity")

here and elsewhere